packages/block-library/src/navigation/use-api-fetch.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Function that fetches data using apiFetch, and updates the status.
+ *
+ * @param {Function} setIsRequesting function to set requesting state to true or false.
+ * @param {Function} setData Function to set data being returned, or empty array on error.
+ * @param {string} path Query path.
+ */
+export default function useApiFetch( setIsRequesting, setData, path ) {
```

Comment:
Instead of setIsRequesting and setData why not an API like this:

```
const { isLoading, data, error } = useApiFetch( queryObject )
```

similar to https://github.com/marcin-piela/react-fetching-library
I've used this on a side project and it was very nice.

Reply:
I have a question. What if I need to make my fetch call cancelable? As far as I know the only way to do that is via something like this

https://github.com/WordPress/gutenberg/blob/ddd3fffdd327d2f9c5202481345fb7427d4a822b/packages/block-editor/src/components/link-control/index.js#L62

How could I mirror that behaviour with this hook?